cmds/exp/nvme_unlock/nvme_unlock.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// The nvme_unlock command is used to unlock a NVMe drive with a
+// HSS-derived password, and rescan the drive to enumerate the
+// unlocked partitions.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"syscall"
+	"unsafe"
+
+	"github.com/u-root/u-root/pkg/hsskey"
+	"github.com/u-root/u-root/pkg/mount/block"
+)
+
+const (
+	opalLockUnlockIoctl = 1092120797
+)
+
+type opalKey struct {
+	lr     byte
+	keyLen byte
+	align  [6]byte
+	key    [256]byte
+}
+
+type opalSessionInfo struct {
+	sum     uint32
+	who     uint32
+	opalKey opalKey
+}
+
+type opalLockUnlock struct {
+	session opalSessionInfo
+	lState  uint32
+	align   [4]byte
+}
+
+var (
+	disk               = flag.String("disk", "/dev/nvme0n1", "The disk to be unlocked")
+	verbose            = flag.Bool("d", false, "print debug output")
+	verboseNoSanitize  = flag.Bool("dangerously-disable-sanitize", false, "Print sensitive information - this should only be used for testing!")
+	noRereadPartitions = flag.Bool("no-reread-partitions", false, "Only attempt to unlock the disk, don't re-read the partition table.")
+	lock               = flag.Bool("lock", false, "Lock instead of unlocking")
+)
+
+func verboseLog(msg string) {
+	if *verbose {
+		log.Print(msg)
+	}
+}
+
+func getSysfsInfo(index string, field string) (string, error) {
+	data, err := os.ReadFile(fmt.Sprintf("/sys/class/nvme/nvme%s/%s", index, field))
+	if err != nil {
+		return "", err
+	}
+	for i, b := range data {
```

Comment:
my reading of this code is that it's trying to get the first string of space-separated text.  It's currently checking for ` ` and `\n`.  What about tab characters? or cr or lf?  Perhaps you can use `strings.Fields` (https://pkg.go.dev/strings#Fields) to split by any whitespace character?  And it might make sense to call `strings.TrimSpace` (https://pkg.go.dev/strings#TrimSpace) if you're worried about leading or trailing spaces too.

Reply:
Fixed, `strings.TrimSpace` seems appropriate for this.